src/Protocol/ECommerce.php

```diff
@@ -252,8 +264,8 @@ protected function generateSignature(array $data, $encoding = 'UTF-8')
             );
 
             $data['receipt_no'] = ProtocolHelper::mbStrPad($data['receipt_no'], 6, "0", STR_PAD_LEFT, $encoding);
+            $data['msgdata'] = ProtocolHelper::mbStrPad($data['msgdata'], 40, " ", STR_PAD_RIGHT, $encoding, $data['respcode'] === self::PAYMENT_RESPONSE_ABORT);
```

Comment:
Could be solved this way too. Although it feels more hacky to me.
Only I would add one additional check:

```
$data['msgdata'] = ProtocolHelper::mbStrPad(
    $data['respcode'] === self::PAYMENT_RESPONSE_ABORT && strlen($data['msgdata']) == 0 ? ' ' : $data['msgdata'],
    40,
    " ",
    STR_PAD_RIGHT,
    $encoding
);
```